### PR TITLE
Feature/8 syntax instance additions

### PR DESCRIPTION
**Pull request by vmarquez**
This supersedes my previous PR.  This is simply some more macros for getting no cost 'syntax' as well as some instance additions.

### CODE REVIEW

**Comment by vmarquez** [comment]
@aloiscochard nothing tricky here but touches a lot of stuff so hopefully can get it merged in quick.

**Comment by vmarquez** [comment]
Pretty basic stuff, but doesn't look like anyone has had time to give it a review. @TomasMikula want to give it a glance before I merge?

File: base/src/main/scala/typeclass/Profunctor.scala [hunk 1]
```diff
@@ -5,6 +5,7 @@ trait Profunctor[F[_, _]] {
   def lmap[A, B, C](fab: F[A, B])(ca: C => A): F[C, B]
   def rmap[A, B, C](fab: F[A, B])(bc: B => C): F[A, C]
   def dimap[A, B, C, D](fab: F[A, B])(ca: C => A)(bd: B => D): F[C, D]
+  def dimapy[A, B, C, D](fab: F[A, B])(ca: C => A, bd: B => D): F[C, D] = ???
```
Comment (TomasMikula) [inline base/src/main/scala/typeclass/Profunctor.scala]: Unintentional `???`?

File: base/src/main/scala/Prelude.scala [hunk 2]
```diff
@@ -55,13 +61,23 @@ trait Prelude  extends data.DisjunctionFunctions
...
     new TraversableSyntax.Ops(ta)
 
+  implicit def PProfunctorOps[P[_,_], A, B](pab: P[A, B])(implicit P: Profunctor[P]): ProfunctorSyntax.Ops[P, A, B] = 
+    new ProfunctorSyntax.Ops(pab)
+  
+  implicit def PChoiceOps[P[_,_], A, B](pab: P[A, B])(implicit P: Choice[P]): ChoiceSyntax.Ops[P, A, B] =
+   new ChoiceSyntax.Ops(pab)
+
+  implicit def PStrongOps[P[_,_], A, B](pab: P[A, B])(implicit P: Strong[P]): StrongSyntax.Ops[P, A, B] =
+    new StrongSyntax.Ops(pab)
```
Comment (TomasMikula) [inline base/src/main/scala/Prelude.scala]: The naming convention set above seemed to be `PprofunctorOps`, `PchoiceOps`, `PstrongOps`.

**Pull request merged**